### PR TITLE
Implement file streaming to accept large files

### PR DESCRIPTION
**Pull request by smoskalenko**
@Magomogo let's discuss the changes/enhances.
Currently I'm testing it also locally for big files / resizing options for images, etc.

### CODE REVIEW

**Comment by smoskalenko** [comment]
@Magomogo FYI, I would upgrade also PHPUnit to fix the error (https://github.com/sebastianbergmann/phpunit/issues/3728) in php 7.4 CI run.

But also looks like [barberry-interfaces/composer.json](https://github.com/Magomogo/barberry-interfaces/blob/master/composer.json) currently also doesn't support PHP8. 

Looks like we also should adjust barberry-interfaces to support PHP8, to be able to run barberry on PHP8 in the future.

File: composer.json [hunk 1]
```diff
@@ -10,16 +10,18 @@
         ]
     },
     "require": {
+        "php": ">=7.3",
         "ext-json": "*",
-        "php": "^5.6 || ^7.0",
-        "barberry/interfaces": "^2.5"
+        "barberry/interfaces": "dev-master as 2.7.x-dev",
+        "guzzlehttp/psr7": "^2.3",
+        "symfony/http-foundation": "^5.4"
```
[the remark this answers is not in the document]
Comment (smoskalenko) [reply composer.json:17]: Hello @Magomogo, initially I’ve added guzzlehttp/psr-7 implementation:

- to accept uploaded files as `UploadedFile` instance with access to file as a stream
- Use `Utils::streamFor` / `Utils::tryFopen` from Guzzle to work with files as a streams in File Storage (method getById()) and in Cache service 

Then I’ve checked if we can use Response from GuzzleHttp to send headers /  respond with content of a file a s stream, etc, and found that it has quite primitive implementation. I’ve remembered that Symfony has `StreamedResponse` and checked it, and it fully fitted to my needs, so I’ve used it for:

- Using StreamedResponse to stream content of a file on GET /{id} request 
- And use Response implementation also for the methods like POST /, DELETE /{id}

Theoretically we can get rid off usage of `symfony/http-foundation` usage and create our own implementation of streamed response based on `GuzzleHttp\Psr7\Response` class, if you wish let’s do it this way or suggest other option.

[the remark this answers is not in the document]
Comment (smoskalenko) [reply composer.json:17]: @Magomogo also I'm not sure that `symfony/http-foundation` based on PSR-7, I would say it more like layer for HTTP specification. And there is special PSR-7 bridge  for this - https://symfony.com/doc/current/components/psr7.html

BTW, my idea to use `symfony/http-foundation` is that we have `StreamedRresponse`/`Response` implementation from a box that are ready to use.

[the remark this answers is not in the document]
Comment (Magomogo) [reply composer.json:17]: Ok, I see, thanks for the explanation.

I'm afraid of loosing the simplicity we have, that made possible to use Barberry for years without any upgrade/service. Now, having these dependencies, we could get some issues, what do you think?

[the remark this answers is not in the document]
Comment (smoskalenko) [reply composer.json:17]: @Magomogo actually barberry complexity/logic was not changed, we've just tuned incoming file uploads/response generation. I hope these known libraries will work without any issues  

BTW today I've tested/checked integration of new barberry (and php client) from feature branches locally, and looks like all known to me functionality works good:

- resize of images 
- conversion to different image formats
- fetch files larger than php `memory_limit`
- upload files larger than `memory_limit`

[the remark this answers is not in the document]
Comment (Magomogo) [reply composer.json:17]: Ok, let's live with it.

**Pull request merged**